### PR TITLE
chore(tests): async await instead of setTimeout

### PR DESCRIPTION
**Pull request by justinmchase**
added a util/wait function and modified a test to use that instead of setTimeout.

This change is extracted from PR #495

### CODE REVIEW

File: lib/util.js [hunk 1]
```diff
@@ -44,6 +44,10 @@ exports.getLabels = function (labelNames, args) {
 	return acc;
 };
 
+exports.wait = async function (ms) {
```
Comment (SimenB) [inline lib/util.js:47]: I don't think we need to expose this out, it's enough to use it in the single test

File: lib/util.js [hunk 2]
```diff
@@ -44,6 +44,10 @@ exports.getLabels = function (labelNames, args) {
...
+	return await new Promise(resolve => setTimeout(resolve, ms), ms);
```
Comment (SimenB) [inline lib/util.js:48]:
```suggestion
	await new Promise(resolve => setTimeout(resolve, ms), ms);
```

**Review by SimenB — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -27,13 +27,18 @@ project adheres to [Semantic Versioning](http://semver.org/).
 
 - The `processResources` metric was added, which keeps a track of all sorts of
   active resources. It consists of the following gauges:
+
```
Comment (justinmchase) [inline CHANGELOG.md:30]: I didn't add this line, for some reason the commit hook is adding it.

**Review by SimenB — APPROVED** [review]
thanks!

File: CHANGELOG.md (hunk 3 above)
Comment (SimenB) [reply CHANGELOG.md]: probably prettier thinking the newline is... prettier 😀

**Pull request merged**